packages/flutter_tools/lib/src/observatory.dart

```diff
@@ -54,16 +74,30 @@ class Observatory {
     _getEventController(data['streamId']).add(event);
   }
 
+  Future<IsolateRef> waitHasIsolate() {
```

Comment:
The name does not imply that it returns the first isolate when it becomes available. How about this:

```dart
Future<IsolateRef> get firstIsolate {
  ..
}
```